gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/ChannelPool.java

```diff
@@ -526,6 +542,8 @@ public <RequestT, ResponseT> ClientCall<RequestT, ResponseT> newCall(
   static class ReleasingClientCall<ReqT, RespT> extends SimpleForwardingClientCall<ReqT, RespT> {
     @Nullable private CancellationException cancellationException;
     final Entry entry;
+    private final AtomicBoolean wasClosed = new AtomicBoolean();
+    private final AtomicBoolean wasReleased = new AtomicBoolean();
```

Comment:
Why both bools?

Reply:
So that we can identify different scenarios next time it happens
1. Multiple calls to onClose()
2. onClose() being called first, try-catch of `start()` being called second
3. try-catch of `start()` being called first, onClose() being called second.

With only one flag, I don't think we can identify these different scenarios.